aips/aip-x.md

```diff
@@ -0,0 +1,237 @@
+
+
+## Impact
+
+This would be a full compatible change to the framework. Existing on chain move modules wouldn't be impacted. We will add those constraints to the Aptos Framework around popular asset types, i.e: Coin/FungibleAsset/Object/NFT.
+
+## Alternative Solutions
+
+Another way to enforce permission is to leverage AIP-56 the runtime access control engine in Move. However, the runtime access control is more suitable to provide coarse grained permissions around modules but it's hard to represent permissions such as withdrawing X APT from a signer. However, we are open to encoperate the changes to the Move on Aptos language extensions in the future if we want to extend the permission system to the general public.
+
+
+## Specification and Implementation Details
+Currently, a signer is a single wrapper around an address to represent that the signature of the transaction passes the address authentication logic. We would like to extend the notion of existing signers in the Move VM so that we could provide APIs to manage contextual data  embedded in such extended signer.
+
+### Lifecycle of a permissioned handle
+
+The lifecycle of the new signer will look like the following:
+
+```jsx
+module aptos_framework::permissioned_signer {
+    struct PermissionedHandle {
+        master_addr: address,
+        permission_addr: address,
+    }
+
+    public fun create_permissioned_handle(
+        master: &signer,
+    ): PermissionedHandle;
+
+    public fun signer_from_permissioned(p: &PermissionedHandle): signer;
+
+    public fun destroy_permissioned_handle(p: PermissionedHandle);
+
+    public fun is_permissioned_signer(s: &signer): bool;
+}
+```
+
+We would introduce a new struct called `PermissionedHandle` that is derived from the original signer. The handle is generated by `create_permissioned_handle_with_expiration`  API, which takes an original signer. In that handle, we will generate a fresh address using auid API to store the permission information that would be needed by the management API.
+
+Once we have the handle, we can then invoke `signer_from_permissioned`  to obtain a permissioned signer value. The invariant would be that the address of the permissioned signer should be exactly the same as the existing master signer that is used to derive the handle. With the expiration date, this API would abort if the wall clock time has exceeded the expiration date. Also the handle should not be storable. The storable variant will need to follow the logic of account delegation. The handle is also not droppable, so that we can enforce the destruction logic by using `destroy_permissioned_handle`  API.
+
+Note that this handle does not have store capability. There will be storable variant of this handle as well but that would contain an expiry time and the creation of this handle can only be invoked in the Aptos Framework as this is equivalent to signer delegation and thus need special security measure.
```

Comment:
hm here we are saying that expiration is a separate concern, but above we treat it together.

probably worth rephrasing - `create_permissioned_handle` only creates a handle within the current transaction, and is not storable.

and then separate section for storable handle with expiration.